### PR TITLE
Incorporate execution statesync changes

### PR DESCRIPTION
**Pull request by omegablitz**

### CODE REVIEW

**Comment by omegablitz** [comment]
Tested on the statesync boxes

**Comment by omegablitz** [comment]
Note to self: wait for monad-crypto/monad#848 to hit master before merging

File: monad-proto/proto/message.proto [hunk 1]
```diff
@@ -66,10 +66,17 @@ message ProtoStateSyncRequest {
...
 }
 
+enum ProtoStateSyncUpsertType {
+  CODE = 0;
+  ACCOUNT = 1;
+  STORAGE = 2;
+  ACCOUNT_DELETE = 3;
+  STORAGE_DELETE = 4;
+}
+
```
Comment (michael-yxchen) [inline monad-proto/proto/message.proto:76]: protobuf recommends reserving 0 for an unspecified value. It's used as the default value when deserializing an unknown enum.

ref: https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum

Comment (omegablitz) [reply monad-proto/proto/message.proto:76]: You're right, but I'm lazy and will fix this later when we add versioning support for state-sync messages

**Pull request merged**